File: react-native/src/config/screenshotMaskLevel.ts
/**
 * The level of masking to apply to the screenshot.
 */
export enum ScreenshotMaskLevel {
  /**
   * The strictest level of masking which masks all text,
   * input fields, images, and videos.
   */
  allTextAndMedia = "all_text_and_media",

  /**
   * Masks all text and input fields, including clickable elements.
   */
  allText = "all_text",

  /**
   * Masks all text and input fields, excluding clickable elements.
   */
  allTextExceptClickable = "all_text_except_clickable",

  /**
   * The most lenient masking level which only masks sensitive input fields
   * like passwords, email fields, and phone number fields.
   */
  sensitiveFieldsOnly = "sensitive_fields_only"
}

File: android/measure/src/main/java/sh/measure/android/config/ScreenshotMaskLevel.kt
package sh.measure.android.config

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * The level of masking to apply to the screenshot.
 */
@Serializable
internal enum class ScreenshotMaskLevel {
    /**
     * The strictest level of masking which masks all text, input fields, images and videos.
     */
    @SerialName("all_text_and_media")
    AllTextAndMedia,

    /**
     * Masks all text and input fields, including clickable elements.
     */
    @SerialName("all_text")
    AllText,

    /**
     * Masks all text and input fields, excluding clickable elements.
     */
    @SerialName("all_text_except_clickable")
    AllTextExceptClickable,

    /**
     * The most lenient level of masking which only masks sensitive input fields like passwords,
     * email and phone number fields.
     */
    @Suppress("unused")
    @SerialName("sensitive_fields_only")
    SensitiveFieldsOnly,
}
